Live Reload stopped working since 3.2.0.CR1

### Describe the bug

Here is the full stacktrace:

```
2023-07-10 11:48:15,947 ERROR [io.qua.dep.dev.IsolatedDevModeMain] (vert.x-worker-thread-1) Failed to start quarkus [Error Occurred After Shutdown]: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.deployment.steps.MainClassBuildStep#build threw an exception: java.lang.RuntimeException: Unable to serialize objects of type class java.util.ImmutableCollections$ListN to bytecode as it has no default constructor
	at io.quarkus.deployment.recording.BytecodeRecorderImpl$46.createValue(BytecodeRecorderImpl.java:1534)
	at io.quarkus.deployment.recording.BytecodeRecorderImpl$DeferredArrayStoreParameter$1.write(BytecodeRecorderImpl.java:2021)
	at io.quarkus.deployment.recording.BytecodeRecorderImpl$FixedMethodContext.writeInstruction(BytecodeRecorderImpl.java:2140)
	at io.quarkus.deployment.recording.BytecodeRecorderImpl$SplitMethodContext.writeInstruction(BytecodeRecorderImpl.java:2097)
	at io.quarkus.deployment.recording.BytecodeRecorderImpl$DeferredArrayStoreParameter.doPrepare(BytecodeRecorderImpl.java:2018)
	at io.quarkus.deployment.recording.BytecodeRecorderImpl$DeferredParameter.prepare(BytecodeRecorderImpl.java:1971)
	at io.quarkus.deployment.recording.BytecodeRecorderImpl$47.doPrepare(BytecodeRecorderImpl.java:1554)
	at io.quarkus.deployment.recording.BytecodeRecorderImpl$DeferredParameter.prepare(BytecodeRecorderImpl.java:1971)
	at io.quarkus.deployment.recording.BytecodeRecorderImpl.writeBytecode(BytecodeRecorderImpl.java:554)
	at io.quarkus.deployment.steps.MainClassBuildStep.writeRecordedBytecode(MainClassBuildStep.java:479)
	at io.quarkus.deployment.steps.MainClassBuildStep.build(MainClassBuildStep.java:261)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:282)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
	at java.base/java.lang.Thread.run(Thread.java:1623)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)

	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:336)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.reloadExistingApplication(AugmentActionImpl.java:266)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.reloadExistingApplication(AugmentActionImpl.java:60)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.restartApp(IsolatedDevModeMain.java:196)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.restartCallback(IsolatedDevModeMain.java:179)
	at io.quarkus.deployment.dev.RuntimeUpdatesProcessor.doScan(RuntimeUpdatesProcessor.java:541)
	at io.quarkus.deployment.dev.RuntimeUpdatesProcessor.doScan(RuntimeUpdatesProcessor.java:441)
	at io.quarkus.vertx.http.runtime.devmode.VertxHttpHotReplacementSetup$4.handle(VertxHttpHotReplacementSetup.java:152)
	at io.quarkus.vertx.http.runtime.devmode.VertxHttpHotReplacementSetup$4.handle(VertxHttpHotReplacementSetup.java:139)
	at io.vertx.core.impl.ContextBase.lambda$null$0(ContextBase.java:137)
	at io.vertx.core.impl.ContextInternal.dispatch(ContextInternal.java:264)
	at io.vertx.core.impl.ContextBase.lambda$executeBlocking$1(ContextBase.java:135)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:1623)
Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.deployment.steps.MainClassBuildStep#build threw an exception: java.lang.RuntimeException: Unable to serialize objects of type class java.util.ImmutableCollections$ListN to bytecode as it has no default constructor
	at io.quarkus.deployment.recording.BytecodeRecorderImpl$46.createValue(BytecodeRecorderImpl.java:1534)
	at io.quarkus.deployment.recording.BytecodeRecorderImpl$DeferredArrayStoreParameter$1.write(BytecodeRecorderImpl.java:2021)
	at io.quarkus.deployment.recording.BytecodeRecorderImpl$FixedMethodContext.writeInstruction(BytecodeRecorderImpl.java:2140)
	at io.quarkus.deployment.recording.BytecodeRecorderImpl$SplitMethodContext.writeInstruction(BytecodeRecorderImpl.java:2097)
	at io.quarkus.deployment.recording.BytecodeRecorderImpl$DeferredArrayStoreParameter.doPrepare(BytecodeRecorderImpl.java:2018)
	at io.quarkus.deployment.recording.BytecodeRecorderImpl$DeferredParameter.prepare(BytecodeRecorderImpl.java:1971)
	at io.quarkus.deployment.recording.BytecodeRecorderImpl$47.doPrepare(BytecodeRecorderImpl.java:1554)
	at io.quarkus.deployment.recording.BytecodeRecorderImpl$DeferredParameter.prepare(BytecodeRecorderImpl.java:1971)
	at io.quarkus.deployment.recording.BytecodeRecorderImpl.writeBytecode(BytecodeRecorderImpl.java:554)
	at io.quarkus.deployment.steps.MainClassBuildStep.writeRecordedBytecode(MainClassBuildStep.java:479)
	at io.quarkus.deployment.steps.MainClassBuildStep.build(MainClassBuildStep.java:261)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:282)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
	at java.base/java.lang.Thread.run(Thread.java:1623)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)

	at io.quarkus.builder.Execution.run(Execution.java:123)
	at io.quarkus.builder.BuildExecutionBuilder.execute(BuildExecutionBuilder.java:79)
	at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:160)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:332)
	... 18 more
Caused by: java.lang.RuntimeException: Unable to serialize objects of type class java.util.ImmutableCollections$ListN to bytecode as it has no default constructor
	at io.quarkus.deployment.recording.BytecodeRecorderImpl$46.createValue(BytecodeRecorderImpl.java:1534)
	at io.quarkus.deployment.recording.BytecodeRecorderImpl$DeferredArrayStoreParameter$1.write(BytecodeRecorderImpl.java:2021)
	at io.quarkus.deployment.recording.BytecodeRecorderImpl$FixedMethodContext.writeInstruction(BytecodeRecorderImpl.java:2140)
	at io.quarkus.deployment.recording.BytecodeRecorderImpl$SplitMethodContext.writeInstruction(BytecodeRecorderImpl.java:2097)
	at io.quarkus.deployment.recording.BytecodeRecorderImpl$DeferredArrayStoreParameter.doPrepare(BytecodeRecorderImpl.java:2018)
	at io.quarkus.deployment.recording.BytecodeRecorderImpl$DeferredParameter.prepare(BytecodeRecorderImpl.java:1971)
	at io.quarkus.deployment.recording.BytecodeRecorderImpl$47.doPrepare(BytecodeRecorderImpl.java:1554)
	at io.quarkus.deployment.recording.BytecodeRecorderImpl$DeferredParameter.prepare(BytecodeRecorderImpl.java:1971)
	at io.quarkus.deployment.recording.BytecodeRecorderImpl.writeBytecode(BytecodeRecorderImpl.java:554)
	at io.quarkus.deployment.steps.MainClassBuildStep.writeRecordedBytecode(MainClassBuildStep.java:479)
	at io.quarkus.deployment.steps.MainClassBuildStep.build(MainClassBuildStep.java:261)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:282)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
	at java.base/java.lang.Thread.run(Thread.java:1623)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)

2023-07-10 11:48:15,958 INFO  [io.qua.dep.dev.RuntimeUpdatesProcessor] (vert.x-worker-thread-1) Live reload total time: 4.142s 

```

### Output of `java -version`

20

### Quarkus version or git rev

3.2.0.CR1 and 3.2.0.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

gradle 8.2

--- a/extensions/oidc/deployment/src/main/java/io/quarkus/oidc/deployment/devservices/AbstractDevConsoleProcessor.java
+++ b/extensions/oidc/deployment/src/main/java/io/quarkus/oidc/deployment/devservices/AbstractDevConsoleProcessor.java
@@ -1,6 +1,7 @@
 package io.quarkus.oidc.deployment.devservices;
 
 import java.time.Duration;
+import java.util.List;
 import java.util.Map;
 
 import jakarta.inject.Singleton;
@@ -109,7 +110,7 @@ public abstract class AbstractDevConsoleProcessor {
             ConfigurationBuildItem configurationBuildItem,
             String keycloakAdminUrl,
             Map<String, String> keycloakUsers,
-            Object keycloakRealms,
+            List<String> keycloakRealms,
             boolean alwaysLogoutUserInDevUiOnReload) {
         final CardPageBuildItem cardPage = new CardPageBuildItem();
 

--- a/extensions/oidc/deployment/src/main/java/io/quarkus/oidc/deployment/devservices/keycloak/KeycloakDevConsoleProcessor.java
+++ b/extensions/oidc/deployment/src/main/java/io/quarkus/oidc/deployment/devservices/keycloak/KeycloakDevConsoleProcessor.java
@@ -1,5 +1,6 @@
 package io.quarkus.oidc.deployment.devservices.keycloak;
 
+import java.util.List;
 import java.util.Map;
 import java.util.Optional;
 
@@ -84,6 +85,9 @@ public class KeycloakDevConsoleProcessor extends AbstractDevConsoleProcessor {
 
             String keycloakAdminUrl = configProps.get().getConfig().get("keycloak.url");
 
+            @SuppressWarnings("unchecked")
+            final List<String> keycloakRealms = (List<String>) configProps.get().getProperties().get("keycloak.realms");
+
             CardPageBuildItem cardPageBuildItem = createProviderWebComponent(
                     recorder,
                     capabilities,
@@ -102,7 +106,7 @@ public class KeycloakDevConsoleProcessor extends AbstractDevConsoleProcessor {
                     configurationBuildItem,
                     keycloakAdminUrl,
                     users,
-                    configProps.get().getProperties().get("keycloak.realms"),
+                    keycloakRealms,
                     configProps.get().isContainerRestarted());
 
             // Also add Admin page

--- a/extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/devui/OidcDevJsonRpcService.java
+++ b/extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/devui/OidcDevJsonRpcService.java
@@ -3,6 +3,7 @@ package io.quarkus.oidc.runtime.devui;
 import static io.quarkus.oidc.runtime.devui.OidcDevServicesUtils.getTokens;
 
 import java.time.Duration;
+import java.util.List;
 import java.util.Map;
 
 import io.quarkus.arc.Arc;
@@ -30,7 +31,7 @@ public class OidcDevJsonRpcService {
     private final String oidcGrantType;
     private final boolean introspectionIsAvailable;
     private final String keycloakAdminUrl;
-    private final Object keycloakRealms;
+    private final List<String> keycloakRealms;
     private final boolean swaggerIsAvailable;
     private final boolean graphqlIsAvailable;
     private final String swaggerUiPath;

--- a/extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/devui/OidcDevUiRecorder.java
+++ b/extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/devui/OidcDevUiRecorder.java
@@ -1,6 +1,7 @@
 package io.quarkus.oidc.runtime.devui;
 
 import java.time.Duration;
+import java.util.List;
 import java.util.Map;
 import java.util.function.Supplier;
 
@@ -11,7 +12,7 @@ public class OidcDevUiRecorder {
     public Supplier<OidcDevUiRpcSvcPropertiesBean> prepareRpcServiceProperties(String authorizationUrl, String tokenUrl,
             String logoutUrl, Duration webClientTimeout, Map<String, Map<String, String>> grantOptions,
             Map<String, String> oidcUsers, String oidcProviderName, String oidcApplicationType, String oidcGrantType,
-            boolean introspectionIsAvailable, String keycloakAdminUrl, Object keycloakRealms, boolean swaggerIsAvailable,
+            boolean introspectionIsAvailable, String keycloakAdminUrl, List<String> keycloakRealms, boolean swaggerIsAvailable,
             boolean graphqlIsAvailable, String swaggerUiPath, String graphqlUiPath, boolean alwaysLogoutUserInDevUiOnReload) {
         return new Supplier<OidcDevUiRpcSvcPropertiesBean>() {
             @Override

--- a/extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/devui/OidcDevUiRpcSvcPropertiesBean.java
+++ b/extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/devui/OidcDevUiRpcSvcPropertiesBean.java
@@ -1,6 +1,7 @@
 package io.quarkus.oidc.runtime.devui;
 
 import java.time.Duration;
+import java.util.List;
 import java.util.Map;
 import java.util.UUID;
 
@@ -17,7 +18,7 @@ public class OidcDevUiRpcSvcPropertiesBean {
     private final String oidcGrantType;
     private final boolean introspectionIsAvailable;
     private final String keycloakAdminUrl;
-    private final Object keycloakRealms;
+    private final List<String> keycloakRealms;
     private final boolean swaggerIsAvailable;
     private final boolean graphqlIsAvailable;
     private final String swaggerUiPath;
@@ -33,7 +34,7 @@ public class OidcDevUiRpcSvcPropertiesBean {
             Duration webClientTimeout, Map<String, Map<String, String>> grantOptions,
             Map<String, String> oidcUsers, String oidcProviderName, String oidcApplicationType,
             String oidcGrantType, boolean introspectionIsAvailable, String keycloakAdminUrl,
-            Object keycloakRealms, boolean swaggerIsAvailable, boolean graphqlIsAvailable,
+            List<String> keycloakRealms, boolean swaggerIsAvailable, boolean graphqlIsAvailable,
             String swaggerUiPath, String graphqlUiPath, boolean alwaysLogoutUserInDevUiOnReload) {
         this.authorizationUrl = authorizationUrl;
         this.tokenUrl = tokenUrl;
@@ -95,7 +96,7 @@ public class OidcDevUiRpcSvcPropertiesBean {
         return keycloakAdminUrl;
     }
 
-    Object getKeycloakRealms() {
+    List<String> getKeycloakRealms() {
         return keycloakRealms;
     }
 

--- a/extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/devui/OidcDevUiRuntimePropertiesDTO.java
+++ b/extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/devui/OidcDevUiRuntimePropertiesDTO.java
@@ -1,5 +1,7 @@
 package io.quarkus.oidc.runtime.devui;
 
+import java.util.List;
+
 import org.eclipse.microprofile.config.Config;
 
 import io.quarkus.oidc.runtime.OidcConfigPropertySupplier;
@@ -26,7 +28,7 @@ public class OidcDevUiRuntimePropertiesDTO {
     private final String oidcGrantType;
     private final boolean introspectionIsAvailable;
     private final String keycloakAdminUrl;
-    private final Object keycloakRealms;
+    private final List<String> keycloakRealms;
     private final boolean swaggerIsAvailable;
     private final boolean graphqlIsAvailable;
     private final String swaggerUiPath;
@@ -36,7 +38,7 @@ public class OidcDevUiRuntimePropertiesDTO {
 
     OidcDevUiRuntimePropertiesDTO(String authorizationUrl, String tokenUrl, String logoutUrl, Config config, int httpPort,
             String oidcProviderName, String oidcApplicationType, String oidcGrantType,
-            boolean introspectionIsAvailable, String keycloakAdminUrl, Object keycloakRealms,
+            boolean introspectionIsAvailable, String keycloakAdminUrl, List<String> keycloakRealms,
             boolean swaggerIsAvailable, boolean graphqlIsAvailable, String swaggerUiPath,
             String graphqlUiPath, boolean alwaysLogoutUserInDevUiOnReload, String propertiesStateId) {
         this.clientId = new OidcConfigPropertySupplier(CLIENT_ID_CONFIG_KEY).get(config);
@@ -114,7 +116,7 @@ public class OidcDevUiRuntimePropertiesDTO {
         return keycloakAdminUrl;
     }
 
-    public Object getKeycloakRealms() {
+    public List<String> getKeycloakRealms() {
         return keycloakRealms;
     }
 

